recipes/pyjion/meta.yaml

```
@@ -0,0 +1,79 @@
+{% set name = "pyjion" %}
+{% set version = "1.2.6" %}
+
+package:
+  name: {{ name|lower }}
+  version: {{ version }}
+
+source:
+  git_url: https://github.com/tonybaloney/Pyjion.git
+  git_tag: {{ version }}
```

`git_url` is strongly discouraged (the bot doesn't understand it, no hashing to make sure the tag doesn't change, ...)

```suggestion
  url: https://github.com/tonybaloney/Pyjion/archive/refs/tags/{{ version }}.tar.gz
  sha256: 1d76194037114d674eada64bea2c04ab276880cf5da4325be186e07d7ce700d2
```

---

I am aware of that, however, this was done specifically because the build uses git submodules which IIUC are not included in the GitHub release archives.

---

IIUC the only workaround is to include the submodule as a separate source download and to unpack it *into* the `pyjion` source folder.

You will need to find out the exact commit hash used and AFAIK this also isn't supported by the bot so it seems to add a lot of complexity and manual work for no gain?

---

> *no hashing to make sure the tag doesn't change*

If the concern is mutable tag references you can reference the actual commit instead I think?

---

There's a fair bit of discussion about what to do in the case of submodules - e.g.
* https://github.com/conda-forge/xgboost-feedstock/issues/22#issuecomment-475972221
* https://github.com/conda-forge/conda-forge.github.io/issues/619#issuecomment-507393863
* https://github.com/conda-forge/coolprop-feedstock/pull/8/files#r331113300

From my point of view, if the bot can't correctly update the submodule source, there isn't much point in bending over backwards to use it. Even if the submodule hash referred to a tag, you don't actually want the bot greedily updating the submodule if it hasn't been updated in the actual repo.

If the bot won't work either way, I'm happy to manually put in PRs to update this package and I think using the `git_url` will incur the lowest maintenance burden.

---

Ah right, I hadn't realised submodules were being used. Yeah in that case I'm happy enough with it using the tag. I'd rather avoid risking the version and commit rev get out of sync.